var/spack/repos/builtin/packages/node-js/package.py

```diff
@@ -103,6 +103,11 @@ class NodeJs(Package):
     # See https://github.com/nodejs/node/issues/52223
     patch("fix-old-glibc-random-headers.patch", when="^glibc@:2.24")
 
+    # Work around gcc-12.[1-2] compiler bug
+    # See https://github.com/nodejs/node/pull/53728
+    # and https://github.com/nodejs/node/issues/53633
+    patch("fix-broken-gcc12-pr53728.patch", when="@22.2:22.5")
```

Comment:
On reflection, Should `gcc` be included in the when clause since it is gcc-specific *and* other compilers could be used?

Reply:
The patch is only adding a preprocessor directive `(__GNUC__ == 12 && __GNUC_MINOR__ <= 2)` so it would not affect other compilers. I usually try to apply patches without conditions so they are automatically tested when new versions are added, though I do understand that is not a good argument here since the version range is closed. Nevertheless, I think I'd prefer setting the example of not adding a compiler in `when`. Your call.